doc/vlan/SONiC_VLAN_trunk_all_except_none_HLD.md

```diff
@@ -0,0 +1,701 @@
+<br> Result: port tagged in vlan 1-4094
+<br> Verify: 'show run-configuration' output.  
+```
+sonic# show running-configuration interface Ethernet X
+!
+interface EthernetX
+ mtu 9100
+ speed 40000
+ shutdown
+ switchport trunk allowed Vlan 1-4094
+sonic#
+```
+**Scenario 15:**
+<br> Step 1: sonic(conf-if-EthernetX)#switchport trunk allowed vlan all
+<br> Step 2: sonic(conf-if-EthernetX)#switchport trunk allowed vlan 1-20
+<br> Result: Port tagged in Vlan1-20
+<br> Verify: 'show run-configuration' output.  
+```
+sonic# show running-configuration interface Ethernet X
+!
+interface EthernetX
+ mtu 9100
+ speed 40000
+ shutdown
+ switchport trunk allowed Vlan 1-20
+sonic#
+```
+
+**Scenario 16:**
+<br> Step 1: sonic(conf-if-EthernetX)#switchport trunk allowed vlan all
+<br> Step 2: sonic(conf-if-EthernetX)#switchport trunk allowed vlan 1-20
+<br> Step 3: sonic(conf-if-EthernetX)#switchport access vlan 20
+<br> Result: Port untagged in Vlan20
+<br> Step 4: sonic(conf-if-EthernetX)#no switchport access vlan 
+<br> Result: Port tagged in vlan1-20 
+<br> Verify: 'show run-configuration' output.  
+
+**Scenario 17**:
+<br> Step 1: sonic(conf-if-EthernetX)#switchport trunk allowed vlan 1-20
+<br> Step 2: sonic(conf-if-EthernetX)#switchport trunk allowed vlan all
+<br> Result: Port tagged in vlan1-4094 
+<br> Verify: 'show run-configuration' output.  
+
+**Scenario 18:**
+<br> Step 1: sonic(conf-if-EthernetX)#switchport trunk allowed vlan except 20-30
+<br> Step 2: sonic(conf-if-EthernetX)#switchport trunk allowed vlan 20-30
+<br> Result: Port tagged in vlan20-30 
+<br> Verify: 'show run-configuration' output.  
+
+**Scenario 19:**
+<br> Step 1: sonic(conf-if-EthernetX)#switchport trunk allowed vlan 20-30
+<br> Step 2: sonic(conf-if-EthernetX)#switchport trunk allowed vlan except 20-30
+<br> Result: Port tagged in vlan 1-19,31-4094
+<br> Verify: 'show run-configuration' output.  
+
+**Scenario 20:**
+<br> Step 1: sonic(conf-if-EthernetX)#switchport trunk allowed vlan all
+<br> Step 2: sonic(conf-if-EthernetX)#switchport trunk allowed vlan remove 1-30
+<br> Result: port tagged in vlan31-4094
+<br> Verify: 'show run-configuration' output.  
+
+**Scenario 21:**
+<br> Step 1: sonic(conf-if-EthernetX)#switchport trunk allowed vlan except 1-10
+<br> Step 2: sonic(conf-if-EthernetX)#switchport trunk allowed vlan remove 11 
+<br> Result: Port tagged in Vlan12-4094
+<br> Verify: 'show run-configuration' output.  
+
+**Scenario 22:**
+<br> Vlan 20 does not exist.
+<br> Step 1: sonic(conf-if-EthernetX)#switchport trunk allowed vlan all
+<br> Step 2: sonic(conf-if-EthernetX)#switchport access vlan 20
+<br> Step 3: sonic(config)#interface vlan 20     //only Vlan 20 existing
+<br> Result: untagged port in vlan 20
+<br> Step 4: sonic(conf-if-EthernetX)#no switchport access vlan
+<br> Result: tagged port in vlan 20
+<br> Step 5: sonic(conf-if-EthernetX)#no switchport trunk allowed vlan 20 
+<br> Result: port will not have any vlan membership
```

Comment:
port will be member to non-existing vlans i.e. 1-19,21-4094 and show running will have this info, right

Reply:
yes, have updated test scenario 5 with show output